File: oop/src/mvp/view/info/Info.java
package mvp.view.info;

import java.util.Collection;

public class Info implements InterfaceForInfo{
       private Object object;

    public Info() {
        this.object =null;
    }

    @Override
    public void show(Object object) {
            this.object=object;
    }

    @Override
    public String toString() {
        if (object instanceof String) {
            System.out.println(object);
            return null;
        } else if (object instanceof Number) {
            System.out.println(object);
            return null;
        } else if (object instanceof Collection<?> collection) {
            for (Object item : collection) {
                System.out.println(item);
            }
            return null;
        } else {
            System.out.println("Объект в неизвестном виде="+object);
            return null;
        }
    }

}

File: oop_dz/src/refactor_by_oop/mvp/view/info/Info.kt
package refactor_by_oop.mvp.view.info

class Info:IInfo {
    private var `object`: Any? = null



    override fun infoShow(`object`: Any?) {
        this.`object` = `object`
    }

    override fun toString(): String {
        if (`object` is String) {
            println(`object`)
            return ""
        } else if (`object` is Number) {
            println(`object`)
            return ""
        } else if (`object` is Collection<*>) {
            for (item in `object` as Collection<*>) {
                System.out.println(item)
            }
            return ""
        } else {
            println("Объект в неизвестном виде=$`object`")
            return ""
        }
    }
}